launch_testing_ros/examples/README.rst

```diff
@@ -0,0 +1,57 @@
+
+* ``hello_world_launch_test.py`` : Demonstrates basic usage and code flow
+  This test attempts to execute the ``echo hello_world`` command as a separate process, checks that the text ``hello_world`` was detected in the stdout of the launched process (in test_read_stdout function), shuts down the original ``echo hello_world`` process if it is still running. 
+  Finally it checks if the shutdown was successful with exit code 0, in the function test_exit_codes.
+* ``check_node_launch_test.py`` : There might be situations where nodes, once launched, take some time to actually start and we need to wait for the node to start to perform some action. We can simulate this using ``launch.actions.TimerAction``. This example shows one way to detect when a node has been launched. We delay the launch by 5 seconds, and wait for the node to start with a timeout of 8 seconds.
+* ``check_msgs_launch_test.py`` : Consider a problem statement where you need to launch a node and check if messages are published on a particular topic. This example demonstrates how to do that, using the talker node from ``demo_nodes_cpp`` package. It uses the ``Event`` object to end the test as soon as the first message is received on the chatter topic, with a timeout of 5 seconds. It is recommended to keep the tests as short as possible.
+* ``set_param_launch_test.py`` : Launch a node, set a parameter in it and check if that was successful.
+
+Running the tests
+-----------------
+Make sure you have your ROS workspace sourced and then run:
+
+``launch_test hello_world_launch_test.py``
+
+The output should be similar to:
+::
+  [INFO] [launch]: All log files can be found below /home/aditya/.ros/log/2021-08-23-09-47-06-613035-aditya-desktop-1376669
+  [INFO] [launch]: Default logging verbosity is set to INFO
+  test_read_stdout (hello_world.TestHelloWorldProcess) ... [INFO] [echo-1]: process started with pid [1376672]
+  ok
+  
+  [INFO] [echo-1]: process has finished cleanly [pid 1376672]
+  ----------------------------------------------------------------------
+  Ran 1 test in 0.008s
+  
+  OK
+  test_exit_codes (hello_world.TestHelloWorldShutdown) ... ok
+  
+  ----------------------------------------------------------------------
+  Ran 1 test in 0.000s
+  
+  OK
+
+Similarly, the other tests can be run after making sure the corresponding dependencies are installed.
+
+Generic examples
+----------------
+
+More ``launch_testing`` examples which are not ROS specific are available `here <https://github.com/ros2/launch/tree/master/launch_testing/test/launch_testing/examples>`_.
+
+* ``good_proc_launch_test.py`` : TODO : Add description
+* ``parameters_launch_test.py`` : TODO : Add desscription
+* ``ready_action_test.py``: TODO : Add description
+* ``terminating_proc_launch_test.py``: TODO : Add description
+* ``args_launch_test.py``: TODO : Add description
+* ``context_launch_test.py``: TODO : Add description
```

Comment:
Instead of adding a description of those examples here, we could link to a readme in the launch/launch_testing.

Reply:
Done